prebuild/emscripten/glue.cpp

```diff
@@ -0,0 +1,538 @@
+  return self->sqr();
+}
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_mul_1(blst::PT* self, const blst::PT* p) {
+  return self->mul(*p);
+}
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_final_exp_0(blst::PT* self) {
+  return self->final_exp();
+}
+
+bool EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_in_group_0(blst::PT* self) {
+  return self->in_group();
+}
+
+bool EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_finalverify_2(blst::PT* self, const blst::PT* gt1, const blst::PT* gt2) {
+  return self->finalverify(*gt1, *gt2);
+}
+
+blst::PT* EMSCRIPTEN_KEEPALIVE emscripten_bind_PT_one_0(blst::PT* self) {
+  blst::PT *one = (blst::PT *) malloc(sizeof(blst::PT));
+  *one = self-> one();
+  return one;
+}
+
+void EMSCRIPTEN_KEEPALIVE emscripten_bind_PT___destroy___0(blst::PT* self) {
+  delete self;
+}
+
+// Pairing
+
+blst::Pairing* EMSCRIPTEN_KEEPALIVE emscripten_bind_Pairing_Pairing_3(bool hash_or_encode, const blst::byte* DST, long DST_len) {
+  return new blst::Pairing(hash_or_encode, DST, DST_len);
+}
+
+long EMSCRIPTEN_KEEPALIVE emscripten_bind_Pairing_aggregate_6__P1_P2(blst::Pairing* self, const blst::P1_Affine* pk, const blst::P2_Affine* sig, const blst::byte* msg, long msg_len, const blst::byte* aug, long aug_len) {
+  return self->aggregate(pk, sig, msg, msg_len, aug, aug_len);
+}
+
+long EMSCRIPTEN_KEEPALIVE emscripten_bind_Pairing_aggregate_6__P2_P1(blst::Pairing* self, const blst::P2_Affine* pk, const blst::P1_Affine* sig, const blst::byte* msg, long msg_len, const blst::byte* aug, long aug_len) {
+  return self->aggregate(pk, sig, msg, msg_len, aug, aug_len);
+}
+
+long EMSCRIPTEN_KEEPALIVE emscripten_bind_Pairing_mul_n_aggregate_8__P1_P2(blst::Pairing* self, const blst::P1_Affine* pk, const blst::P2_Affine* sig, const blst::byte* scalar, long scalar_len, const blst::byte* msg, long msg_len, const blst::byte* aug, long aug_len) {
+  return self->mul_n_aggregate(pk, sig, scalar, scalar_len, msg, msg_len, aug, aug_len);
+}
+
+long EMSCRIPTEN_KEEPALIVE emscripten_bind_Pairing_mul_n_aggregate_8__P2_P1(blst::Pairing* self, const blst::P2_Affine* pk, const blst::P1_Affine* sig, const blst::byte* scalar, long scalar_len, const blst::byte* msg, long msg_len, const blst::byte* aug, long aug_len) {
+  return self->mul_n_aggregate(pk, sig, scalar, scalar_len, msg, msg_len, aug, aug_len);
+}
+
+void EMSCRIPTEN_KEEPALIVE emscripten_bind_Pairing_commit_0(blst::Pairing* self) {
+  self->commit();
+}
+
+long EMSCRIPTEN_KEEPALIVE emscripten_bind_Pairing_merge_1(blst::Pairing* self, const blst::Pairing* ctx) {
+  return self->merge(ctx);
+}
+
+bool EMSCRIPTEN_KEEPALIVE emscripten_bind_Pairing_finalverify_1(blst::Pairing* self, const blst::PT* sig) {
+  return self->finalverify(sig);
+}
+
+void EMSCRIPTEN_KEEPALIVE emscripten_bind_Pairing___destroy___0(blst::Pairing* self) {
+  delete self;
+}
+
+}
+
+blst::P1* EMSCRIPTEN_KEEPALIVE emscripten_bind_G1_0() {
+  blst::P1 *p1 = (blst::P1 *) malloc(sizeof(blst::P1));
+  *p1 = blst::G1();
+
+  return p1;
```

Comment:
```suggestion
  return reinterpret_cast<blst::P1*>(new blst::G1());
```

Reply:
I think your suggestion is semantically different. We need to return a `P1*` (pointer) but `G1()` returns a `P1` (value). Here is something that I got the compiler to be happy with but I'm not sure if there's any other fancy C++ machinery that I'm not aware that can make it more idiomatic or concise:

```
  blst::P1* g1 = new blst::P1();
  *g1 = blst::G1();
  return g1;

```

Reply:
Okay yes have re-read this and you are correct `G1` and `G2` are just inline functions. This make this easier then. We can combine both lines above as:

```c++
return new blst::P1(blst::G1());
```

This will construct a new copy (on the heap) from the generated `G1()` value